packages/mask/src/plugins/IdeaMarket/SNSAdaptor/ListingsView.tsx

```diff
@@ -0,0 +1,364 @@
+                const value = Number(params.value) * 100
+                const operator = value >= 0 ? '+' : ''
+                return `${operator}${value.toFixed()}%`
+            },
+        },
+        {
+            field: 'price',
+            headerName: t('plugin_ideamarket_price'),
+            type: 'number',
+            width: 64,
+            headerAlign: 'center',
+            align: 'center',
+            valueFormatter: (params: GridValueFormatterParams) => formatterToUSD.format(params.value as number),
+        },
+        {
+            field: 'deposits',
+            headerName: t('plugin_ideamarket_deposits'),
+            headerAlign: 'center',
+            width: 95,
+            type: 'number',
+            align: 'center',
+            valueFormatter: (params: GridValueFormatterParams) => formatterToUSD.format(params.value as number),
+        },
+        {
+            field: 'button',
+            headerName: '',
+            sortable: false,
+            width: 80,
+            align: 'center',
+            renderCell: (params: GridRenderCellParams) => (
+                <Grid container alignContent="center" justifyContent="center">
+                    <Grid item>
+                        <Button
+                            href={`${BASE_URL}/i/${params.row.id}`}
+                            target="_blank"
+                            color="primary"
+                            size="small"
+                            variant="contained">
+                            {t('plugin_ideamarket_buy')}
+                        </Button>
+                    </Grid>
+                </Grid>
+            ),
+        },
+    ]
+
+    function CustomLoading() {
+        return (
+            <div className={classes.empty}>
+                <LoadingAnimation />
+            </div>
+        )
+    }
+
+    function NoRowsCustomOverlay() {
+        return (
+            <Stack height="100%" alignItems="center" justifyContent="center">
+                {t('no_data')}
+            </Stack>
+        )
+    }
+
+    function NoResultsCustomOverlay() {
+        return (
+            <Stack height="100%" alignItems="center" justifyContent="center">
+                {t('no_data')}
+            </Stack>
+        )
+    }
+
+    function handlePageChange(newPage: number) {
+        setPage(newPage)
+    }
+
+    return (
+        <div className={classes.root}>
+            <DataGrid
+                className={classes.grid}
+                paginationMode="server"
+                rowsPerPageOptions={[20]}
+                page={page}
+                pageSize={20}
+                onPageChange={handlePageChange}
+                headerHeight={18}
+                components={{
+                    Toolbar: QuickSearchToolbar,
+                    LoadingOverlay: CustomLoading,
+                    NoResultsOverlay: NoResultsCustomOverlay,
+                    NoRowsOverlay: NoRowsCustomOverlay,
```

Comment:
Those components should be defined outside of this component.

Reply:
https://github.com/DimensionDev/Maskbook/pull/5924#discussion_r867997089

Reply:
Therefore I encapsulated api token, see: 696067d8c0f9be7d090c5c82c983043f98c3eb34